test/app-tap/datetime.test.lua

```diff
@@ -1316,6 +1319,54 @@ test:test("Time intervals creation - range checks", function(test)
         local err_msg, attribs = unpack(row)
         assert_raises(test, err_msg, function() return inew(attribs) end)
     end
+
+    local boundary_values = {
+        {{ year = -MAX_YEAR_RANGE }, '-11759221 years'},
+        {{ year =  MAX_YEAR_RANGE }, '+11759221 years'},
+        {{ month = -MAX_MONTH_RANGE }, '-141110652 months'},
+        {{ month =  MAX_MONTH_RANGE }, '+141110652 months'},
+        {{ week = -MAX_WEEK_RANGE }, '-613579352 weeks'},
+        {{ week =  MAX_WEEK_RANGE }, '+613579352 weeks'},
+        {{ day = -math.floor(MAX_DAY_RANGE) }, '-4295055470 days'},
+        {{ day =  math.floor(MAX_DAY_RANGE) }, '+4295055470 days'},
+        {{ hour = -MAX_HOUR_RANGE }, '-103081331286 hours'},
+        {{ hour =  MAX_HOUR_RANGE }, '+103081331286 hours'},
+        {{ min = -MAX_MIN_RANGE }, '-6184879877160 minutes'},
+        {{ min =  MAX_MIN_RANGE }, '+6184879877160 minutes'},
+        {{ sec = -MAX_SEC_RANGE }, '-371092792629600 seconds'},
+        {{ sec =  MAX_SEC_RANGE }, '+371092792629600 seconds'},
+        {{ nsec = -MAX_NSEC_RANGE }, '+0 seconds, -2147483647 nanoseconds'},
+        {{ nsec =  MAX_NSEC_RANGE }, '+0 seconds, 2147483647 nanoseconds'},
```

Comment:
Boundary values in messages are calculated manually and hardcoded now. I would build these message in runtime:
```
'-11759221 years' -> ('-%d'):format(MAX_YEAR_RANGE)
'+11759221 years' -> ('%d'):format(MAX_YEAR_RANGE) 
...
```
In latter case these messages will depend on real boundary values and automatically will be updated when we will update boundary values.

Reply:
Actually, same goes for +-1 test, it seems quite easy to automate and make it more compact as well.

Reply:
Reworked tests to be the most compact and automized

Reply:
Thanks!